### PR TITLE
Fix tsp, add tests on AdditionalProperties

### PR DESCRIPTION
**Pull request by v-hongli1**
Add test cases on AdditionalProperties.

### CODE REVIEW

File: typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        Assertions.assertTrue(record.isFlag());
+        Assertions.assertNotNull(record.getAdditionalProperties());
+        Assertions.assertNotNull(record.getAdditionalProperties().get("prop1"));
+        Assertions.assertEquals("\"abc\"", record.getAdditionalProperties().get("prop1").toString());
+        Assertions.assertNotNull(record.getAdditionalProperties().get("prop2"));
+        Assertions.assertEquals(43.125, Double.parseDouble(record.getAdditionalProperties().get("prop2").toString()));
+    }
+
+    @Test
+    @Disabled("The body provided of 'put' request doesn't match expected body")
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java:200]: This is your test problem.

These should be `BinaryData.fromObject(new WidgetData2())` or `BinaryData.fromObject(new WidgetData1())`

See source https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp#L531-L536

Comment (weidongxu-microsoft) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java:200]: Same for the getter part. It need to be converted to `WidgetData1` with `BinaryData.toObject`

Comment (weidongxu-microsoft) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java:200]: There is a bug in code that make the "put" part not working, though.

Hopefully there be a relatively easy fix.

Comment (weidongxu-microsoft) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java:200]: Fixed in PR https://github.com/Azure/autorest.java/pull/2699

Please refresh main after it merged, and try enable those tests about `BinaryData`.

I've also use one of your test in that PR (with changes). Please check that as well. Basically you should always use `fromObject` and `toObject` on `BinaryData`.

Comment (v-hongli1) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java:204]: Fixed in the new version.

Comment (weidongxu-microsoft) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java:204]: Why is this test still `@Disabled`?

File: typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        Assertions.assertNotNull(record);
+        Assertions.assertEquals("abc", record.getKnownProp());
+        Assertions.assertNotNull(record.getAdditionalProperties());
+        Assertions.assertNotNull(record.getAdditionalProperties().get("prop"));
+        record.getAdditionalProperties().get("prop").forEach(modelForRecord ->
+                Assertions.assertEquals("ok", modelForRecord.getState()));
+    }
+
+    @Test
+    @Disabled("The body provided of 'put' request doesn't match expected body")
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java:184]: What's wrong with this test?

Comment (v-hongli1) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java]: Enabled the test cases except `*ModelArray`.

**Review by weidongxu-microsoft — APPROVED** [review]

File: typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java (hunk 1 above)
Comment (v-hongli1) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java]: Enabled the test cases except `*ModelArray`.

**Pull request merged**